config-handler-router/src/test/groovy/org/codice/admin/router/ConfigurationHandlerRouterTest.groovy

```diff
@@ -0,0 +1,294 @@
+        1 * response.status(400)
+    }
+
+    def 'test persist(Request, Response) config handler but fail persist'() {
+        setup:
+        prepareConfigHandler(MATCHING_ID, ConfigurationHandlerRouter.PERSIST_ID)
+
+        def report = mockReport(true)
+
+        when:
+        configurationHandlerRouter.persist(request, response)
+
+        then:
+        1 * configurationHandler.persist(METHOD_ID, _ as Configuration) >> report
+        1 * response.status(400)
+    }
+
+    def 'test probe(Request, Response) config handler success'() {
+        setup:
+        prepareConfigHandler(MATCHING_ID, ConfigurationHandlerRouter.PROBE_ID)
+
+        def report = Mock(ProbeReport) {
+            containsUnsuccessfulMessages() >> false
+        }
+
+        when:
+        configurationHandlerRouter.probe(request, response)
+
+
+        then:
+        1 * configurationHandler.probe(METHOD_ID, _ as Configuration) >> report
+        0 * response.status(400)
+    }
+
+    def 'test probe(Request, Response) no config handler'() {
+        setup:
+        prepareConfigHandler(NON_MATCHING_ID, ConfigurationHandlerRouter.PROBE_ID)
+
+        when:
+        def report = configurationHandlerRouter.probe(request, response)
+
+        then:
+        report.messages().size() == 1
+        report.messages().get(0).message() == FAILURE_MESSAGE
+        report.messages().get(0).configFieldId() == ConfigurationHandlerRouter.CONFIGURATION_TYPE_FIELD
+        report.containsFailureMessages()
+        0 * configurationHandler.probe(_ as String, _ as Configuration)
+        1 * response.status(400)
+    }
+
+    def 'test probe(Request, Response) config handler but fail persist'() {
+        setup:
+        prepareConfigHandler(MATCHING_ID, ConfigurationHandlerRouter.PROBE_ID)
+
+        def report = Mock(ProbeReport) {
+            containsUnsuccessfulMessages() >> true
+        }
+
+        when:
+        configurationHandlerRouter.probe(request, response)
+
+
+        then:
+        1 * configurationHandler.probe(METHOD_ID, _ as Configuration) >> report
+        1 * response.status(400)
+    }
+
+    def 'test getCapabilities()'() {
+        setup:
+        prepareConfigHandler(NON_MATCHING_ID, ConfigurationHandlerRouter.TEST_ID)
+
+        when:
+        def reports = (List<CapabilitiesReport>)configurationHandlerRouter.getCapabilities()
+
+        then:
+        reports.size() == 1
+        reports.get(0).getConfigurationType() == CONFIGURATION_TYPE
+    }
+
+    def 'test configCapabilties(Request, Response) success'() {
+        setup:
+        prepareConfigHandler(MATCHING_ID, ConfigurationHandlerRouter.TEST_ID)
+
+        when:
+        configurationHandlerRouter.configCapabilities(request, response)
+
+        then:
+        1 * configurationHandler.getCapabilities()
+        configurationHandler.getCapabilities().configurationType == CONFIGURATION_TYPE
+    }
+
+    def 'test configCapabilties(Request, Response) null config handler'() {
```

Comment:
Not sure what this test description means at first glance, might want to be more descriptive. Something like 'test configCapabilties(Request, Response) with no matching config handler'